djsonb/lookups.py

```diff
@@ -72,10 +74,27 @@ def sql(self):
                         sql_tuple = FilterTree.text_similarity_filter(rule[0], pattern, True)
                     else:
                         sql_tuple = FilterTree.text_similarity_filter(rule[0], pattern, False)
-                    rule_specs.append(sql_tuple)
+                    # add to the list of rules generated for this pattern (one per field)
+                    got_pattern = patterns.get(pattern) or []
+                    got_pattern.append(sql_tuple)
+                    patterns[pattern] = got_pattern
+
+        rule_strings = [' AND '.join([rule[0] for rule in rule_specs])]
+
+        pattern_rules = patterns.values()
+        pattern_strings = []
+
+        for rule_list in pattern_rules:
```

Comment:
FWIW it appears things have changed since I read this tip (to not use function calls in the head of a loop unless it's a generator) in an early edition of _Learning Python_ some years ago.